protostream/src/main/java/org/infinispan/protostream/ProtostreamBenchmark.java

```diff
@@ -34,43 +35,70 @@ public class ProtostreamBenchmark {
    @Param({"true", "false"})
    boolean byteArrayOrStream;
 
+   @Benchmark
+   public Object testSizedAddress(ContextState contextState, AddressState addressState) throws IOException {
+      SerializationContext ctx = contextState.getCtx();
+      Address address = addressState.getAddress();
+      TagWriterImpl sizedWriter = TagWriterImpl.newInstance(ctx);
+      WrappedMessage.write(ctx, sizedWriter, address);
+      return sizedWriter;
+   }
+
    @Benchmark
    public Object testMarshallAddress(ContextState contextState, AddressState addressState) throws IOException {
+      RandomAccessOutputStreamImpl raos = contextState.getOutputStream();
       SerializationContext ctx = contextState.getCtx();
       Address address = addressState.getAddress();
-      if (byteArrayOrStream)
-         return ProtobufUtil.toWrappedByteArray(ctx, address);
-      else {
-         ByteArrayOutputStreamEx os = new ByteArrayOutputStreamEx(addressState.getAddressBytes().length);
-         ProtobufUtil.toWrappedStream(ctx, os, address);
-         return os.getByteBuffer();
+      if (byteArrayOrStream) {
+         raos.write(ProtobufUtil.toWrappedByteArray(ctx, address));
+      } else {
+         ProtobufUtil.toWrappedStream(ctx, (RandomAccessOutputStream) raos, address);
```

Comment:
This change is important as the stream version will no longer cause an additional allocation per invocation which was causing it to appear "slower" in some cases.